server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/ImapRequestFrameDecoder.java

```diff
@@ -221,17 +221,21 @@ public void close() throws IOException {
 
     @Override
     protected synchronized ChannelBuffer newCumulationBuffer(ChannelHandlerContext ctx, int minimumCapacity) {
-        @SuppressWarnings("unchecked")
         Map<String, Object> attachment = (Map<String, Object>) ctx.getAttachment();
-        int size = (Integer) attachment.get(NEEDED_DATA);
-        
-        if (inMemorySizeLimit > 0) {
-            return ChannelBuffers.dynamicBuffer(Math.min(size, inMemorySizeLimit), ctx.getChannel().getConfig().getBufferFactory());
-        } else {
+        if (attachment.containsKey(NEEDED_DATA)) {
+            @SuppressWarnings("unchecked")
+            int size = (Integer) attachment.get(NEEDED_DATA);
+
+            if (inMemorySizeLimit > 0) {
+                return ChannelBuffers.dynamicBuffer(Math.min(size, inMemorySizeLimit), ctx.getChannel().getConfig().getBufferFactory());
+            } else {
 
-            if (size > 0) {
-                return ChannelBuffers.dynamicBuffer(size, ctx.getChannel().getConfig().getBufferFactory());
+                if (size > 0) {
+                    return ChannelBuffers.dynamicBuffer(size, ctx.getChannel().getConfig().getBufferFactory());
+                }
+                return super.newCumulationBuffer(ctx, minimumCapacity);
             }
+        } else {
             return super.newCumulationBuffer(ctx, minimumCapacity);
```

Comment:
Is `else` necessary? you could just have that call to the super method at the end, which would avoid repeating the same line 236